### PR TITLE
Gatling: new default VS distributed

### PR DESCRIPTION
**Pull request by mbounabi**
https://jira.kisio.org/browse/NAVP-898

### CODE REVIEW

File: gatling/simulations/navitia/distributed.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ object Base {
         .baseURL("http://localhost:5000/v1/coverage") //jormun
         .acceptEncodingHeader("gzip, deflate")
 
-  def query(scn: String) = {
+  def query(scn: String, first_mode: String, last_mode:String) = {
```
Comment (xlqian) [inline gatling/simulations/navitia/distributed.scala]: first/last_mode should be an array

File: gatling/simulations/navitia/distributed.scala [hunk 2]
```diff
@@ -18,15 +18,60 @@ object Base {
           .queryParam("to", "${to}")
           .queryParam("datetime", "${datetime}")
           .queryParam("datetime_represents", "${datetime_represents}")
+          .queryParam("first_section_mode[]", first_mode)
```
Comment (xlqian) [inline gatling/simulations/navitia/distributed.scala]: we should be able to take several first_mode into count

File: gatling/readme.md [hunk 3]
```diff
@@ -1,8 +1,65 @@
...
+callable = app
+processes = 4
+lazy-apps = True
+gevent = 4
+enable-threads = True
+```
+
+**Run Jormun with uWSGI**
+
+Active the jormungandr virtual env and in *root_project/source/jormungandr/jormaungandr* run :
```
Comment (pbougue) [inline gatling/readme.md]: Activ`at`e (or `Enable` or `Use`)

File: gatling/simulations/navitia/distributed.scala [hunk 4]
```diff
@@ -18,15 +18,60 @@ object Base {
...
 }
 
-class NewDefault extends Simulation {
-    setUp(scenario("new_default").exec(Base.query("new_default")).inject(constantUsersPerSec(1) during(300) randomized).protocols(Base.httpConf))
+class Distributed_Walking_Bike extends Simulation {
+    setUp(scenario("Distributed_Walking_Bike").exec(Base.query("distributed", "walking", "bike"))
+      .inject(constantUsersPerSec(1) during(300) randomized).protocols(Base.httpConf))
 }
+
+class Distributed_Bike_Walking extends Simulation {
```
Comment (pbougue) [inline gatling/simulations/navitia/distributed.scala]: not necessarily in this PR, I would group scenario a bit, following 2 dimensions:
* NewDefault vs. Distributed
* `WalkOnly` vs. `CombinedModes` vs. `SameJourneySchedules` (testing timeframe and all that calls that we are going to serve more often)... and maybe vs. `WeirdParams` (if necessary)

**Review by pbougue — APPROVED** [review]
Needs to be run and tested anyway (before or after merge doesn't matter)

File: gatling/simulations/navitia/distributed.scala (hunk 4 above)
Comment (mbounabi) [reply gatling/simulations/navitia/distributed.scala]: There's also an alternative to have several config files and handle fallback modes with the config file loaded. This solution will abstract the scala part of Gatling testing

**Comment by benoit-bst** [comment]
The last commit concentrate all config tests requests in **csv file**.
You just need to complete .csv if you want to test a new Request.

```csv
coverage,type,from,to,datetime,datetime_represents,first_section_mode,last_section_mode,other_args

fr-idf,walking-walking departure,2.37715;48.846781,2.43425;48.794755,20180710T090000,departure,walking,walking,
fr-idf,walking-walking arrival,2.37715;48.846781,2.43425;48.794755,20180710T100000,arrival,walking,walking,
fr-idf,walking-walking min_nb=3,2.37715;48.846781,2.43425;48.794755,20180710T090000,departure,walking,walking,min_nb_journeys=3
fr-idf,walking-walking count=10,2.37715;48.846781,2.43425;48.794755,20180710T090000,departure,walking,walking,count=10

etc...
```

Moreover, that allow to launch only : **distributed** or **new default** with all Requests data set :

term

### The gatling output is now a html like this :

gatling_main

### if you zoon within a particular request you have :+1: 

gatling_zoom

**Review by benoit-bst — APPROVED** [review]

**Review by xlqian — APPROVED** [review]

**Pull request merged**